backend/src/services/openai.ts

```diff
@@ -10,6 +10,13 @@ export class OpenAIService {
         this.openai = new OpenAI({ apiKey });
     }
 
+    private logMessage(label: string, log: any, endMessage?: string): void {
```

Comment:
I pulled this out to hopefully make `analyzePlaylist` a bit easier to read and easier to wire up to a proper logging system (like data dog). Let me know if you prefer it the old way, I'm happy to revert if this makes it worse for you.